snapshots/devmapper/snapshotter.go

```diff
@@ -428,17 +490,21 @@ func (s *Snapshotter) getDevicePath(snap storage.Snapshot) string {
 	return dmsetup.GetFullDevicePath(name)
 }
 
-func (s *Snapshotter) buildMounts(snap storage.Snapshot) []mount.Mount {
+func (s *Snapshotter) buildMounts(ctx context.Context, snap storage.Snapshot, fileSystemType fsType) []mount.Mount {
 	var options []string
 
+	if fileSystemType == "" {
+		log.G(ctx).Error("File system type cannot be empty")
```

Comment:
why not just return error? just in case

Reply:
This is to support existing config of users. If containerd is upgraded to use this change, but config is still not updated to have a specified file system type, then we assume that its ext4 as that was the only supported file system type. If config has a file system type entry, then it will be used. So technically this is to add backward compatibility. This should also cover the case when user has the filesystem type in their configuration but leaves it empty. We will fallback to ext4 that way too.

Reply:
The suggestion regarding adding empty check in validate() looks good. Fixing it in next version.